test/parallel/test-assert.js

```diff
@@ -501,6 +501,12 @@ assert.throws(
   }
 );
 
+a.equal(NaN, NaN);
+a.throws(
+  () => a.notEqual(NaN, NaN),
+  a.AssertionError
+);
```

Comment:
Is this missing `notDeepEqual` test or is it somewhere else?

Reply:
It's tested in https://github.com/nodejs/node/pull/30766/files#diff-b22a683d7d673587e547a5adb452fc1fR598-R600

`assertDeepAndStrictEqual` is a helper to test both sides of the loose and strict version.

Reply:
As far as I can tell it doesn't explicitly test the `throw` of `notDeepEqual` (the same way `a.notEqual(NaN, NaN)` is tested here)
```javascript
function assertDeepAndStrictEqual(a, b) {
  assert.deepEqual(a, b);
  assert.deepStrictEqual(a, b);

  assert.deepEqual(b, a);
  assert.deepStrictEqual(b, a);
}
```

I meant to have explicit
`common.expectsError(() => a.notDeepEqual(NaN, NaN), { ... });`

Reply:
True. We could add tests for that as well. The internal functionality only uses a single implementation though and if deep equal is truthy, the negation can't be. That is not the case with `.equal()` and `.notEqual()` since these do not share a common implementation.

Would you rather add tests for this case as well?

Reply:
Well, I'd rather not depend on the fact that internally they use the same implementation and add explicit tests as the tests will stay even when the internals [non-breakingly] change.
Though this is non-blocking anyway.